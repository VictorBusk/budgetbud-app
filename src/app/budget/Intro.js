import React, {Component} from 'react';
import {
    Button,
    Container,
    Label,
    Spinner,
    Text
} from "native-base";
import {AppHeader, ConfirmDialog} from "../../components";
import {Image} from "react-native";
import {connect} from "react-redux";
import {color, container, text, button} from "../../style";
import I18n from "../../strings/i18n";
import {getLinkedAccounts} from "../../redux/actions";

class Intro extends Component {
    componentWillMount() {
        this.props.getLinkedAccounts();
    }

    render() {
        return (
            <Container style={[container.signedInContainer]}>
                <ConfirmDialog
                    title={I18n.t('introConfirmDialogHeader')}
                    text={I18n.t('introConfirmDialogBody')}
                    confirmCallback={() => {
                        this.props.navigation.navigate("Accounts");
                        this.confirmDialog.dismissDialog();
                    }}
                    budgetLoading={this.props.budgetLoading}
                    ref={(confirmDialog) => {
                        this.confirmDialog = confirmDialog
                    }}
                />

                <AppHeader headerText={I18n.t('introHeader')}
                           onLeftButtonPress={() => this.props.navigation.navigate("DrawerOpen")}
                />
                <Container style={container.justifyCenter}>
                    {this.props.accountsLoading ? (
                        <Spinner style={{
                            alignItems: 'center',
                            justifyContent: 'center'
                        }} color='#1c313a'/>) : (
                        <Container style={container.defaultFormStyle}>
                            <Container style={{flex: 1,}}>
                                <Container style={{flex: 0.15, justifyContent: 'flex-end'}}>
                                    <Text style={[text.headerText, color.text]}>
                                        {I18n.t('introCreateBudgetHeader')}
                                    </Text>
                                </Container>
                                <Container style={{
                                    flex: 0.7,
                                    alignItems: 'center',
                                    justifyContent: 'center'
                                }}>
                                    <Image style={{width: 220, height: 220, borderRadius: 110, borderColor: '#03426A', borderWidth: 2}}
                                           source={require('../../../assets/logo.png')}/>
                                </Container>

                                <Container style={{flex: 0.15, justifyContent: 'flex-start', alignItems: 'flex-end'}}>
                                    <Text style={[text.bodyText, container.bodyContainer, color.text]}>
                                        {I18n.t('introCreateBudgetBody')}
                                    </Text>
                                </Container>
                            </Container>
                            <Button rounded
                                    onPress={() => this.props.linkedAccounts.length === 0 ?
                                        (this.confirmDialog.showDialog()) :
                                        (this.props.navigation.navigate("CreateBudget"))}
                                    style={[button.defaultButton, color.button, {width: '100%'}]}
                            >
                                {this.props.budgetLoading ? (
                                    <Spinner color='#D0D0D0'/>) : (
                                    <Label style={color.white}>
                                        {I18n.t('introGetStarted')}
                                    </Label>
                                )}
                            </Button>
                        </Container>)}
                </Container>
            </Container>
        );
    }
}

const mapStateToProps = (state) => {
    const {budgetID} = state.budget;
    const {linkedAccounts, accountsLoading} = state.account;

    return {budgetID, linkedAccounts, accountsLoading};
};

const mapDispatchToProps = {
    getLinkedAccounts
};

export default connect(mapStateToProps, mapDispatchToProps)(Intro);
