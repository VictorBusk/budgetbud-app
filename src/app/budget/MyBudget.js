import React, {Component} from 'react';
import {
    Body,
    Button,
    Container,
    Icon,
    ListItem,
    Text,
    View,
    Label,
    Spinner
} from "native-base";
import {AppHeader, Separator, Toolbox} from "../../components/index";
import {FlatList, StyleSheet} from "react-native";
import {connect} from "react-redux";
import I18n from "../../strings/i18n";
import {
    container
} from "../../style/index";
import {getBudget} from "./budget_actions";
import {getCategories} from "../category/category_actions";
import {getDebts} from "../debt/debt_actions";

class MyBudget extends Component {
    componentWillMount() {
        if (!this.props.budgetInitialized)
            this.props.getBudget(this.props.budgetID);

        if (!this.props.categoriesInitialized)
            this.props.getCategories(this.props.budgetID);

        if (!this.props.debtsInitialized)
            this.props.getDebts(this.props.budgetID);
    }

    navigateUser = (destination) => {
        this.props.navigation.navigate(destination)
    };

    render() {
        return (
            <Container style={container.signedInContainer}>
                <AppHeader headerText={I18n.t('myBudgetHeader')}
                           onLeftButtonPress={
                               () => this.props.navigation.navigate("DrawerOpen")}
                />
                <Container>
                    <View style={{flex: 0.1, justifyContent: 'center'}}>
                        {this.props.budgetLoading ? (
                            <Spinner style={{
                                alignItems: 'center',
                                justifyContent: 'center'
                            }} color='#1c313a'/>) : (
                            <View style={styles.incomeFormStyle}>
                                <Text style={styles.listText}>{I18n.t('budgetIncome')}
                                </Text>
                                <Text style={styles.listText}>{this.props.income} {I18n.t('currency')}
                                </Text>
                            </View>
                        )}
                    </View>
                    <Separator/>

                    <View style={{flex: 0.7}}>
                        {this.props.categoriesLoading ? (
                            <Spinner style={{
                                alignItems: 'center',
                                justifyContent: 'center',
                                flex: 1
                            }} color='#1c313a'/>) : (
                            <View style={{flex: 1, alignSelf: 'stretch', paddingTop: 20}}>
                                <View style={styles.incomeFormStyle}>
                                    <Label style={styles.textStyle}>
                                        {I18n.t('budgetExpenses')}
                                    </Label>
                                </View>
                                <FlatList
                                    data={this.props.categories}
                                    renderItem={this.renderCategory}
                                    keyExtractor={item => item.categoryTypeID}
                                />

                            </View>
                        )}

                        <Separator/>

                        {this.props.debts.length !== 0 &&
                        <View style={this.props.debts.length > 2 ?
                            {flex: 1, alignSelf: 'stretch', paddingTop: 20} :
                            {flex: 0.5, alignSelf: 'stretch', paddingTop: 20}}>
                            <View style={styles.incomeFormStyle}>
                                <Label style={styles.textStyle}>{I18n.t('budgetDebts')}</Label>
                            </View>
                            <FlatList
                                data={this.props.debts}
                                renderItem={this.renderDebt}
                                keyExtractor={item => item.id}
                            />
                            <Separator/>
                        </View>
                        }
                    </View>

                    <View style={{flex: 0.2}}>
                        <View style={styles.budgetSummary}>
                            <View style={[styles.incomeFormStyle, {paddingTop: '5%', flex: 1}]}>
                                <Text style={styles.listText}>
                                    {I18n.t('budgetTotalExpenses')}
                                </Text>
                                <Text style={styles.listText}>
                                    {this.props.totalGoalsAmount} {I18n.t('currency')}
                                </Text>
                            </View>

                            <View style={[styles.incomeFormStyle, {flex: 1}]}>
                                <Text style={styles.listText}>
                                    {I18n.t('budgetDisposable')}
                                </Text>
                                <Text style={styles.listText}>
                                    {this.props.disposable} {I18n.t('currency')}
                                </Text>
                            </View>

                            <Button transparent
                                    onPress={() => this.bottomModal.showModal()}
                                    style={[styles.buttonStyle, {flex: 1}]}
                            >
                                <Icon name="ios-arrow-dropup-circle"
                                      style={{color: "#00263A"}}/>
                            </Button>

                            <Toolbox
                                ref={(bottomModal) => {
                                    this.bottomModal = bottomModal
                                }}
                                navigateUser={this.navigateUser}
                            />
                        </View>
                    </View>
                </Container>
            </Container>
        );
    }

    renderCategory = ({item}) => {
        return (
            <ListItem style={{paddingRight: 18}}>
                <Body>
                <Text style={styles.listText}>{item.name}</Text>
                </Body>
                <Text style={[styles.listText, {justifyContent: 'flex-end'}]}>
                    {item.amount} {I18n.t('currency')}
                </Text>
            </ListItem>
        );
    };

    renderDebt = ({item}) => {
        return (
            <ListItem style={{paddingRight: 18}}>
                <Body>
                <Text style={styles.listText}>{item.debtData.name}</Text>
                </Body>
                <Text style={[styles.listText, {justifyContent: 'flex-end'}]}>
                    {item.debtData.amountPerMonth} {I18n.t('currency')}</Text>
            </ListItem>
        );
    };
}

const styles = StyleSheet.create({
    buttonStyle: {
        justifyContent: 'flex-end',
        alignSelf: 'flex-end',
        paddingBottom: 10
    },
    incomeFormStyle: {
        alignSelf: 'center',
        width: '100%',
        paddingRight: 18,
        paddingLeft: 15,
        justifyContent: 'space-between',
        flexDirection: 'row'
    },
    expenseListStyle: {
        alignSelf: 'center',
        width: '90%',
        justifyContent: 'space-between',
        flexDirection: 'row',
        paddingTop: 23
    },
    itemStyle: {
        fontWeight: '600',
        alignSelf: 'flex-start',
        color: 'white'
    },
    spacedText: {
        flexGrow: 1,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        marginHorizontal: '5%',
        marginVertical: 10
    },
    budgetSummary: {
        marginBottom: 0,
        justifyContent: 'space-between',
        flex: 1
    },
    textStyle: {
        fontWeight: '400',
        fontSize: 14,
        alignSelf: 'flex-start',
        marginLeft: 5,
        color: '#00263A'
    },
    listText: {
        marginLeft: 5,
        alignSelf: 'flex-start',
        fontSize: 16,
        color: '#00263A'
    },
    modalButton: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
    }
});

const mapStateToProps = (state) => {
    const {
        categories,
        categoriesLoading,
        totalGoalsAmount,
        categoriesInitialized
    } = state.category;

    const {
        debts,
        debtsInitialized
    } = state.debt;

    const {
        budgetLoading,
        income,
        destination,
        budgetID,
        budgetInitialized
    } = state.budget;

    const disposable = state.disposable.disposable;

    return {
        categories,
        budgetLoading,
        income,
        debts,
        totalGoalsAmount,
        disposable,
        destination,
        budgetID,
        categoriesLoading,
        budgetInitialized,
        categoriesInitialized,
        debtsInitialized
    }
};

export default connect(mapStateToProps, {
    getBudget, getCategories, getDebts
})(MyBudget);
