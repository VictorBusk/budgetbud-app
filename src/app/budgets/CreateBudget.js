import React, {Component} from 'react';
import {Keyboard} from 'react-native';
import {Container} from 'native-base';
import {connect} from 'react-redux';
import {BudgetForm, AppHeader} from "../../components";
import I18n from "../../strings/i18n";
import {container} from "../../style";
import {
    showWarningToast,
    checkInputAmount,
    correctConversion,
    setupNewCategoriesList
} from "../../helpers";
import {
    createBudget,
    getLinkedAccounts,
    createCategories,
    mapExpensesToBudget
} from "../../redux/actions";
import {resetBudgetError} from "./budget_actions";
import {resetCategoriesError} from "../categories/category_actions";

class CreateBudget extends Component {
    state = {
        tmpIncome: this.props.income,
        tmpDisposable: this.props.disposable,
        tmpTotalGoalsAmount: this.props.totalGoalsAmount,
        tmpCategories: [],
        submitLoading: false
    };

    async componentWillMount() {
        this.setState({
            tmpCategories: await this.props.mapExpensesToBudget()
        })
    };

    componentWillReceiveProps(nextProps) {
        if (nextProps.budgetError) {
            showWarningToast(nextProps.budgetError);
            this.props.resetBudgetError();
        } else if (nextProps.categoriesError) {
            showWarningToast(nextProps.categoriesError);
            this.props.resetCategoriesError();
        }
    }

    onIncomeChange = (newIncome) => {
        newIncome = correctConversion(newIncome);
        if (checkInputAmount(newIncome)) {
            const incomeDiff = newIncome - this.state.tmpIncome;
            const newDisposable = Math.round((this.state.tmpDisposable + incomeDiff) * 100) / 100;

            this.setState({
                tmpIncome: newIncome,
                tmpDisposable: newDisposable,
            })
        }
    };

    onCategoryChange = (name, oldAmount, newAmount) => {
        newAmount = correctConversion(newAmount);
        if (checkInputAmount(newAmount)) {
            const categoryDiff = oldAmount - newAmount;
            const newDisposable = Math.round((this.state.tmpDisposable + categoryDiff) * 100) / 100;
            const newTotalGoalsAmount = Math.round((this.state.tmpTotalGoalsAmount - categoryDiff) * 100) / 100;
            const newTmpCategories = setupNewCategoriesList(
                this.state.tmpCategories,
                name,
                newAmount);

            this.setState({
                tmpDisposable: newDisposable,
                tmpTotalGoalsAmount: newTotalGoalsAmount,
                tmpCategories: newTmpCategories
            })
        }
    };

    handleSubmit = async () => {
        Keyboard.dismiss();
        this.setState({
            submitLoading: true
        });

        const categoryItems = this.state.tmpCategories.filter((obj) => {
            return obj.amount > 0
        });

        await this.props.createBudget(
            this.state.tmpIncome,
            this.state.tmpDisposable,
            this.state.tmpTotalGoalsAmount,
            (budgetID) => {
                this.props.createCategories
                (
                    budgetID,
                    categoryItems,
                    () => {
                        this.setState({
                            submitLoading: false
                        });
                    }
                )
            }
        );
    };

    render() {
        return (
            <Container style={[container.signedInContainer, {alignItems: 'stretch'}]}>
                <AppHeader headerText={I18n.t('createBudgetHeader')}
                           infoText={I18n.t('createBudgetInfo')}
                           showBackButton={true}
                           onLeftButtonPress={() => {
                               if (!this.state.submitLoading)
                                   this.props.navigation.pop()
                           }}/>

                <BudgetForm handleSubmit={this.handleSubmit}
                            onIncomeChanged={this.onIncomeChange}
                            onCategoryChanged={this.onCategoryChange}
                            budgetID={this.props.budgetID}
                            tmpIncome={this.state.tmpIncome}
                            tmpTotalGoalsAmount={this.state.tmpTotalGoalsAmount}
                            tmpDisposable={this.state.tmpDisposable}
                            tmpCategories={this.state.tmpCategories}
                            debts={this.props.debts}
                            budgetLoading={this.props.budgetLoading}
                            submitLoading={this.state.submitLoading}
                            categoriesLoading={this.props.categoriesLoading}
                            budgetError={this.props.budgetError}
                />
            </Container>
        );
    }
}

const mapStateToProps = (state) => {
    const {
        income,
        debts,
        budgetLoading,
        budgetError,
        budgetID
    } = state.budget;

    const disposable = state.disposable.disposable;

    const linkedAccounts = state.account.linkedAccounts;
    const {
        categoriesLoading,
        categoriesError,
        totalGoalsAmount
    } = state.category;

    return {
        income,
        budgetID,
        linkedAccounts,
        debts,
        totalGoalsAmount,
        disposable,
        budgetLoading,
        categoriesError,
        budgetError,
        categoriesLoading
    };
};

const mapDispatchToProps = {
    createBudget,
    getLinkedAccounts,
    mapExpensesToBudget,
    createCategories,
    resetBudgetError,
    resetCategoriesError
};

export default connect(mapStateToProps, mapDispatchToProps)(CreateBudget);
