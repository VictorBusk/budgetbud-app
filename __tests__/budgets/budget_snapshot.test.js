import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CreateBudget from "../../src/app/budgets/CreateBudget";
import EditBudget from "../../src/app/budgets/EditBudget";
import MyBudget from "../../src/app/budgets/MyBudget";
import thunk from 'redux-thunk';
import {
    INITIAL_ACCOUNT_STATE,
    INITIAL_BUDGET_STATE,
    INITIAL_CATEGORY_STATE,
    INITIAL_DEBT_STATE,
    INITIAL_DISPOSABLE_STATE
} from '../test_helper/initial_state'
import {BudgetForm} from "../../src/components/forms/BudgetForm";
import Intro from "../../src/app/budgets/Intro";

Enzyme.configure({adapter: new Adapter()});

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Budget', () => {
    it('renders CreateBudget as expected', () => {
        const wrapper = shallow(
            <CreateBudget/>,
            {
                context: {
                    store: mockStore({
                        budget: INITIAL_BUDGET_STATE,
                        disposable: INITIAL_DISPOSABLE_STATE,
                        account: INITIAL_ACCOUNT_STATE,
                        category: INITIAL_CATEGORY_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders EditBudget as expected', () => {
        const wrapper = shallow(
            <EditBudget/>,
            {
                context: {
                    store: mockStore({
                        budget: INITIAL_BUDGET_STATE,
                        disposable: INITIAL_DISPOSABLE_STATE,
                        account: INITIAL_ACCOUNT_STATE,
                        category: INITIAL_CATEGORY_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders MyBudget as expected', () => {
        const wrapper = shallow(
            <MyBudget/>,
            {
                context: {
                    store: mockStore({
                        budget: INITIAL_BUDGET_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        debt: INITIAL_DEBT_STATE,
                        disposable: INITIAL_DISPOSABLE_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders Intro as expected', () => {
        const wrapper = shallow(
            <Intro/>,
            {
                context: {
                    store: mockStore({
                        budget: INITIAL_BUDGET_STATE,
                        category: INITIAL_CATEGORY_STATE,
                        debt: INITIAL_DEBT_STATE,
                        disposable: INITIAL_DISPOSABLE_STATE,
                        account: INITIAL_ACCOUNT_STATE
                    })
                }
            },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });

    it('renders Budgetform in loading state as expected', () => {
        const wrapper = shallow(
            <BudgetForm tmpTotalGoalsAmount={12} categoriesLoading={true}/>
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});