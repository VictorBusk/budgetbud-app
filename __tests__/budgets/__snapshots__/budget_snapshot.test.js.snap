// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Budget renders Budgetform in loading state as expected 1`] = `
<Container
  style={
    Object {
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "marginBottom": 5,
          "marginTop": 10,
          "width": "90%",
        },
        Object {
          "paddingTop": 10,
        },
      ]
    }
  >
    <Styled(Label)
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "fontSize": 14,
            "fontWeight": "400",
            "marginLeft": 5,
          },
          Object {
            "color": "#003755",
          },
        ]
      }
    >
      Enter income
    </Styled(Label)>
    <Styled(Item)
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "marginVertical": 5,
            "paddingVertical": 0,
          },
          Object {
            "backgroundColor": "#F6F7F9",
            "borderColor": "#003755",
          },
        ]
      }
    >
      <Styled(Input)
        keyboardType="numeric"
        style={
          Object {
            "color": "#003755",
          }
        }
        value="undefined"
      />
    </Styled(Item)>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Separator />
  </View>
  <View
    style={
      Object {
        "flex": 4,
        "justifyContent": "center",
      }
    }
  >
    <Styled(Spinner)
      color="#1c313a"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    />
  </View>
  <Separator />
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": "3%",
            "width": "90%",
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          Total expense goals
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          12
           
          kr
        </Styled(Text)>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": "3%",
            "width": "90%",
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          Disposable
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
           
          kr
        </Styled(Text)>
      </View>
    </View>
  </View>
  <View>
    <Styled(Button)
      onPress={[Function]}
      rounded={true}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "height": 40,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginTop": 20,
            "width": "90%",
          },
          Object {
            "backgroundColor": "#00263A",
          },
        ]
      }
    >
      <Styled(Text)
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontWeight": "600",
          }
        }
      >
        Save
      </Styled(Text)>
    </Styled(Button)>
  </View>
</Container>
`;

exports[`Budget renders CreateBudget as expected 1`] = `
<Styled(Container)
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
      },
      Object {
        "alignItems": "stretch",
      },
    ]
  }
>
  <AppHeader
    headerText="Create budget"
    infoText="Here you can create a new budget. The expense goals can be pre-filled based on your expenses for the last three months from your connected accounts."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <BudgetForm
    budgetError=""
    budgetID=""
    budgetLoading={false}
    categoriesLoading={false}
    handleSubmit={[Function]}
    onCategoryChanged={[Function]}
    onIncomeChanged={[Function]}
    submitLoading={false}
    tmpCategories={Array []}
    tmpDisposable={0}
    tmpIncome={0}
    tmpTotalGoalsAmount={0}
  />
</Styled(Container)>
`;

exports[`Budget renders EditBudget as expected 1`] = `
<Styled(Container)
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
      },
      Object {
        "alignItems": "stretch",
      },
    ]
  }
>
  <AppHeader
    headerText="Edit budget"
    infoText="Here you can edit your budget by updating the amounts in the input fields and pressing [Save]. Expense goals will only be added if they have an amount greater than 0."
    onLeftButtonPress={[Function]}
    showBackButton={true}
  />
  <BudgetForm
    budgetError=""
    budgetID=""
    budgetLoading={false}
    categoriesLoading={false}
    handleSubmit={[Function]}
    onCategoryChanged={[Function]}
    onIncomeChanged={[Function]}
    submitLoading={false}
    tmpCategories={Array []}
    tmpDisposable={0}
    tmpIncome={0}
    tmpTotalGoalsAmount={0}
  />
</Styled(Container)>
`;

exports[`Budget renders Intro as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="My budget"
    infoText="Here you can get an overview of your budget. Press the arrow icon to open the toolbox. The three buttons in the toolbox navigate to editing screens where you can manage parts of your budget."
    onLeftButtonPress={[Function]}
  />
  <Styled(Container)>
    <Styled(ViewNB)
      style={
        Object {
          "flex": 0.1,
          "justifyContent": "center",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 18,
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          Income
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          0
           
          kr
        </Styled(Text)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Separator />
    <Styled(ViewNB)
      style={
        Object {
          "flex": 0.7,
        }
      }
    >
      <SectionList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        renderSectionHeader={[Function]}
        scrollEventThrottle={50}
        sections={
          Array [
            Object {
              "data": Array [],
              "isCategorySection": true,
              "title": "Goals for expenses",
            },
            Object {
              "data": Array [],
              "isCategorySection": false,
              "title": "Goals for debts",
            },
          ]
        }
        stickySectionHeadersEnabled={true}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
      <Separator />
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "flex": 0.2,
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
            "marginBottom": 0,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "flex": 0.6,
              "justifyContent": "space-between",
            }
          }
        >
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                  "paddingTop": "5%",
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Total expense goals
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                  "paddingTop": "4%",
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Disposable
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
        </Styled(ViewNB)>
        <Styled(Button)
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 0.4,
              },
            ]
          }
          transparent={true}
        >
          <Styled(Icon)
            name="ios-arrow-dropup-circle"
            style={
              Object {
                "color": "#00263A",
              }
            }
          />
        </Styled(Button)>
        <Toolbox
          navigateUser={[Function]}
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Container)>
</Styled(Container)>
`;

exports[`Budget renders MyBudget as expected 1`] = `
<Styled(Container)
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <AppHeader
    headerText="My budget"
    infoText="Here you can get an overview of your budget. Press the arrow icon to open the toolbox. The three buttons in the toolbox navigate to editing screens where you can manage parts of your budget."
    onLeftButtonPress={[Function]}
  />
  <Styled(Container)>
    <Styled(ViewNB)
      style={
        Object {
          "flex": 0.1,
          "justifyContent": "center",
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 18,
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          Income
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "fontSize": 16,
                "marginLeft": 5,
              },
              Object {
                "color": "#003755",
              },
            ]
          }
        >
          0
           
          kr
        </Styled(Text)>
      </Styled(ViewNB)>
    </Styled(ViewNB)>
    <Separator />
    <Styled(ViewNB)
      style={
        Object {
          "flex": 0.7,
        }
      }
    >
      <SectionList
        data={Array []}
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onEndReachedThreshold={2}
        renderItem={[Function]}
        renderSectionHeader={[Function]}
        scrollEventThrottle={50}
        sections={
          Array [
            Object {
              "data": Array [],
              "isCategorySection": true,
              "title": "Goals for expenses",
            },
            Object {
              "data": Array [],
              "isCategorySection": false,
              "title": "Goals for debts",
            },
          ]
        }
        stickySectionHeadersEnabled={true}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
      <Separator />
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "flex": 0.2,
        }
      }
    >
      <Styled(ViewNB)
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-between",
            "marginBottom": 0,
          }
        }
      >
        <Styled(ViewNB)
          style={
            Object {
              "flex": 0.6,
              "justifyContent": "space-between",
            }
          }
        >
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                  "paddingTop": "5%",
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Total expense goals
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
          <Styled(ViewNB)
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingLeft": 15,
                  "paddingRight": 18,
                },
                Object {
                  "flex": 1,
                  "paddingTop": "4%",
                },
              ]
            }
          >
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              Disposable
            </Styled(Text)>
            <Styled(Text)
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontSize": 16,
                    "marginLeft": 5,
                  },
                  Object {
                    "color": "#003755",
                  },
                ]
              }
            >
              0
               
              kr
            </Styled(Text)>
          </Styled(ViewNB)>
        </Styled(ViewNB)>
        <Styled(Button)
          onPress={[Function]}
          style={
            Array [
              Object {
                "alignSelf": "flex-end",
                "justifyContent": "flex-end",
              },
              Object {
                "flex": 0.4,
              },
            ]
          }
          transparent={true}
        >
          <Styled(Icon)
            name="ios-arrow-dropup-circle"
            style={
              Object {
                "color": "#00263A",
              }
            }
          />
        </Styled(Button)>
        <Toolbox
          navigateUser={[Function]}
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(Container)>
</Styled(Container)>
`;
